ls1intum/Artemis
# `Assessment`: Fix the loss of submission text due to overlapping manual text blocks

### Checklist
#### General

#### Client

### Motivation and Context
Closes #6903 

### Description
Removed the console error for the case when manual text blocks were overlapping. Instead, added some code to handle the case as follows: determine the block that has a feedback assigned to it and add a new text block, if needed, to ensure that the block of the new feedback does not hijack some of the existing text.

### Steps for Testing
Prerequisites:
- 1 Text Exercise with a submission ready for assessment
- 1 Tutor

1. Log in to Artemis as Tutor
2. Go to Course Management -> Assessment Dashboard -> Exercise Dashboard of the text exercise
5. Start an assessment
6. Select a specific portion of the text, by holding the alt/option key. Let's call the selected range [r1, r2]
7. Add some feedback and save
8. Refresh the page
9. Verify that you see your feedback
10. Delete the feedback
11. Again, hold alt/option key to select a different range of text. BUT, make sure that the new range overlaps with [r1, r2]
12. Add some feedback and save
13. Refresh the page
14. Verify that you can see your new feedback

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
#### Client
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [text-assessment-base.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5187/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/text-assessment-base.component.ts.html) | 84% | ✅ |
